jupyterhub/oauth/store.py

```diff
@@ -52,14 +52,7 @@ def user_has_denied_access(self, request):
 class HubDBMixin(object):
     """Mixin for connecting to the hub database"""
     def __init__(self, session_factory):
-        self.session_factory = session_factory
-        self._local = threading.local()
-
-    @property
-    def db(self):
-        if not hasattr(self._local, 'db'):
-            self._local.db = scoped_session(self.session_factory)()
-        return self._local.db
+        self.db = session_factory()
```

Comment:
This avoids giving OAuth a separate session from the rest of the app (again, related to the test threads that are now gone)